src/jdk.jpackage/macosx/native/common/MacSysInfo.cpp

```diff
@@ -77,4 +80,15 @@ tstring getProcessModulePath() {
     return FileUtils::toAbsolutePath(reply);
 }
 
+tstring_array getCommandArgs(CommandArgProgramNameMode progNameMode) {
+    tstring_array result;
+    for (int i = progNameMode == ExcludeProgramName ? 1 : 0; i < argc; i++) {
+        if (strncmp(argv[i], PSN_ARG, strlen(PSN_ARG)) == 0) {
+            continue;
+        }
+        result.push_back(argv[i]);
+    }
```

Comment:
I'd suggest to use C++ here:
```suggestion
    const tstring psnArgPrefix = "-psn_";
    for (int i = progNameMode == ExcludeProgramName ? 1 : 0; i < argc; i++) {
        const tstring arg = argv[i];
        if (!tstrings::startsWith(arg, psnArgPrefix)) {
            result.push_back(arg);
        }
    }
```